component/loki/source/windowsevent/component_windows.go

```diff
@@ -0,0 +1,148 @@
+package windowsevent
+
+import (
+	"context"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/loki"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/windows"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "loki.source.windowsevent",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+var (
+	_ component.Component = (*Component)(nil)
+)
+
+// Component implements the loki.source.windowsevent component.
+type Component struct {
+	opts component.Options
+
+	mut       sync.RWMutex
+	args      Arguments
+	target    *windows.Target
+	handler   chan api.Entry
+	receivers []loki.LogsReceiver
+}
+
+func (c *Component) Chan() chan<- api.Entry {
+	return c.handler
+}
+
+func (c *Component) Stop() {
+	// This is a noop.
+}
```

Comment:
I'm assuming these implement some kind of interface for `windows.New`? Personally, I'd like us to create a new (unexported) struct for the business logic so implementing the interface for windows.New doesn't become a part of the public API for the component.